src/widget/wfindonmenu.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <QAction>
+#include <QMenu>
+#include <QUrlQuery>
+
+class Track;
+
+class WFindOnMenu : public QMenu {
+    Q_OBJECT
+  public:
+    explicit WFindOnMenu(
+            QWidget* parent = nullptr);
+    ~WFindOnMenu() override = default;
+
+    enum class Services {
+        SoundCloud,
+        LastFm,
+        Discogs
+    };
+
+    void openTheBrowser(const QString& serviceUrl,
```

Comment:
The menu is controlled by signals and should not expose implementation details as public member functions.